src/verify_cert.rs

```diff
@@ -341,11 +336,32 @@ fn check_basic_constraints(
     }
 }
 
+/// Extended Key Usage (EKU) of a certificate.
+pub enum ExtendedKeyUsage {
+    /// The certificate must contain the specified [`KeyPurposeId`] as EKU.
+    Required(KeyPurposeId),
+
+    /// If the certificate has EKUs, then the specified [`KeyPurposeId`] must be included.
+    RequiredIfPresent(KeyPurposeId),
+}
+
+/// An OID value indicating the Extended Key Usage (EKU) of the certificate.
 #[derive(Clone, Copy, PartialEq, Eq)]
-pub(crate) struct KeyPurposeId {
+pub struct KeyPurposeId {
     oid_value: untrusted::Input<'static>,
 }
 
+impl KeyPurposeId {
+    /// Construct a new [`KeyPurposeId`].
+    ///
+    /// `oid` is the OBJECT IDENTIFIER in bytes.
+    pub const fn new(oid: &'static [u8]) -> Self {
```

Comment:
I think now that this is `const` we could also tidy the `EKU_SERVER_AUTH`, `EKU_CLIENT_AUTH`, and `EKU_OCSP_SIGNING` constants in this file to use `KeyPurposeId::new` instead of manually initializing the struct.